gui/src/components/health/cortx-health-graphical.vue

```diff
@@ -0,0 +1,544 @@
+* by the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU Affero General Public License for more details.
+* You should have received a copy of the GNU Affero General Public License
+* along with this program. If not, see <https://www.gnu.org/licenses/>.
+* For any questions about this software or licensing,
+* please email opensource@seagate.com or cortx-questions@seagate.com.
+*/
+<template>
+  <div>
+    <div id="treeContainer"></div>
+  </div>
+</template>
+<script lang="ts">
+import CortxTabs, { TabsInfo } from "../widgets/cortx-tabs.vue";
+import { Component, Vue, Prop } from "vue-property-decorator";
+import { Api } from "../../services/api";
+import apiRegister from "../../services/api-register";
+import * as d3 from "d3";
+// import { tree } from "vued3tree";
+import * as moment from "moment";
+
+@Component({
+  name: "cortx-health-graphical"
+})
+export default class CortxHealthGraphical extends Vue {
+  public healthQueryParams: any = {};
+  public clusterHealthData: any = {};
+
+  public mounted() {
+    this.getHealthData();
+  }
+
+  public async getHealthData() {
+    this.healthQueryParams = {
+      depth: 0
+    };
+    this.$store.dispatch("systemConfig/showLoader", "Fetching health...");
+    const res = await Api.getAll(apiRegister.health_cluster,
+      this.healthQueryParams
+    );
+    this.clusterHealthData = res.data;
+
+    function traverse(obj) {
+      if (obj.sub_resources && obj.sub_resources.length > 0) {
+        rename(obj);
+        if (obj.children.length > 0) {
+          for (let i = 0; i < obj.children.length; i++) {
+            traverse(obj.children[i]);
+          }
+        }
+      } else {
+        rename(obj);
+      }
+    }
+
+    function rename(obj) {
+      obj.children = obj.sub_resources;
+      delete obj.sub_resources;
+    }
+
+    traverse(this.clusterHealthData.data[0]);
+
+    this.$store.dispatch("systemConfig/hideLoader");
+
+    this.plotTree();
+  }
+
+  public plotTree() {
+    // Set the dimensions and margins of the diagram
+    const margin = { top: 20, right: 90, bottom: 30, left: 90 };
+    const width = 1000 - margin.left - margin.right;
+    const height = 500 - margin.top - margin.bottom;
+    // append the svg object to the body of the page
+    // appends a 'group' element to 'svg'
+    // moves the 'group' element to the top left margin
+    const svg = d3
+      .select("#treeContainer")
+      .append("svg")
+      .attr("width", width + margin.right + margin.left)
+      .attr("height", height + margin.top + margin.bottom)
+      .style("overflow", "auto")
+      .append("g")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+    const i = 0,
```

Comment:
Is this variable used anywhere? If not please remove it.

Reply:
Removed this as not used anywhere in a code.